File: gems/espalier/test/fixtures/big_o_langs/typed.cs
using System.Collections.Generic;
using System.Text;

public class Typed {
  public int SumLengths(List<string> items) {
    int total = 0;
    foreach (string item in items) { total += item.Length; }
    return total;
  }

  public string Join(List<string> items) {
    StringBuilder sb = new StringBuilder();
    foreach (string item in items) { sb.Append(item); }
    return sb.ToString();
  }

  public bool Has(Dictionary<string, int> index, string key) {
    return index.ContainsKey(key);
  }

  public List<string> Copy(List<string> items) {
    List<string> output = new List<string>();
    foreach (string item in items) { output.Add(item); }
    return output;
  }
}

File: gems/espalier/test/fixtures/big_o_langs/typed.ts
export function sumLengths(items: string[]): number {
  let total = 0;
  for (const item of items) { total += item.length; }
  return total;
}

export function lookup(index: Map<string, number>, key: string): number {
  const found = index.get(key);
  return found === undefined ? 0 : found;
}

export function copy(items: string[]): string[] {
  const output: string[] = [];
  for (const item of items) { output.push(item); }
  return output;
}
